runtime/file.go

```diff
@@ -44,7 +44,15 @@ type File struct {
 // NewFileFromFD creates a file object from the given file descriptor fd.
 func NewFileFromFD(fd uintptr) *File {
 	// TODO: Use fcntl or something to get the mode of the descriptor.
-	return &File{Object: Object{typ: FileType}, mode: "?", open: true, file: os.NewFile(fd, "<fdopen>")}
+	file := &File{Object: Object{typ: FileType}, mode: "?", open: true, file: os.NewFile(fd, "<fdopen>")}
+	initFileReader(file)
+	return file
+}
+
+func initFileReader(file *File) {
+	if file != nil && file.file != nil {
+		file.reader = bufio.NewReader(file.file)
```

Comment:
Generally I prefer to avoid helper functions unless they're encapsulating a lot of logic or they avoid some duplication. How about putting the file.reader initialization directly into the &File{...} assignment in NewFileFromFD()?